compliance/collection/nodeinventorizer/cachedInventorizer.go

```diff
@@ -0,0 +1,163 @@
+package nodeinventorizer
+
+import (
+	"os"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	cmetrics "github.com/stackrox/rox/compliance/collection/metrics"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+)
+
+// InventoryScanOpts control behaviour of all node inventory related functions
+type InventoryScanOpts struct {
+	NodeName            string
+	Scanner             NodeInventorizer
+	InventoryCachePath  string
+	BackoffFilePath     string
+	BackoffWaitCallback func(time.Duration)
+}
+
+// TriggerNodeInventory creates a Node Inventory to send to Sensor.
+// To reduce strain on the Node, a linear backoff is checked before collecting an inventory.
+// Additionally, a cached inventory from an earlier invocation may be used instead of a full inventory run if it is fresh enough.
+// Note: This does not prevent strain in case of repeated pod recreation, as both mechanisms are based on an EmptyDir.
+func TriggerNodeInventory(opts *InventoryScanOpts) (*sensor.MsgFromCompliance, error) {
+	// check for existing backoff, wait for specified duration if needed, then persist the new backoff duration
+	initialBackoff := env.NodeInventoryInitialBackoff.DurationSetting()
+	currentBackoff, err := getCurrentBackoff(opts)
+	if err != nil {
+		return nil, err
+	}
+	if *currentBackoff > initialBackoff {
+		log.Warnf("Found existing backoff - last scan may have failed. Waiting %v seconds before running next inventory", currentBackoff)
+		opts.BackoffWaitCallback(*currentBackoff)
+	}
+	nextBackoff := calcNextBackoff(*currentBackoff)
+	bErr := writeBackoff(nextBackoff, opts.BackoffFilePath)
+	if bErr != nil {
+		log.Warnf("Error while persisting inventory backoff interval: %v", bErr)
+	}
```

Comment:
Two subjective suggestions.

1. You can move logging `bErr` inside `writeBackoff` function. This will "unload" the reader of `TriggerNodeInventory` a bit.
2. To my taste, the logged message is very internal. I suppose there's very little chance users may know what `inventory` means.
If I exercise, my user optics, I can come up with this:
```go
log.Warnf("Error writing node scan backoff marker: %v", bErr)
```
How about your optics?

Reply:
Sorry, I don't quite believe in blanket improve-all-logging tickets like ROX-14358 in its today's form. First, it's easy to miss things there, second, the PR context is lost by the time that other task is started. I see that you improved most of logged messages, but some things I'd still comment on.

Let us reuse this thread for general discussion about ROX-14358 or similar. And let me close my existing outdated comments and reopen new ones where I see potential for improvement.

I don't mind ROX-14358 but I think we can more efficiently address many things while being in this PR.

If you're concerned about "node inventory"/"node scan" mismatch in the logging, then let's adopt "node scan" here and leave it to ROX-14358 to ensure the consistent language. :-D